pkg/router/route_group.go

```diff
@@ -46,64 +81,134 @@ type RouteGroup struct {
 	done    chan struct{}
 	once    sync.Once
 
-	rt routing.Table
-
-	lastSent int64
+	readTimer  *time.Timer
```

Comment:
For the timeouts, we can use https://golang.org/src/net/pipe.go as @nkryuchkov has pointed out here: https://discordapp.com/channels/466450852098605059/466450852098605061/652522206290771969

Reply:
@evanlinjin I agree, just pushed :)